app/controllers/api/alert_statuses_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Api
+  class AlertStatusesController < BaseController
+    include Subcollections::AlertStatusStates
+    def options
+      payload = {"providers" => []}
+      ExtManagementSystem.select do |m|
+        m.kind_of?(ManageIQ::Providers::CloudManager) || m.kind_of?(ManageIQ::Providers::InfraManager) || m.kind_of?(ManageIQ::Providers::ContainerManager)
+      end.each do |provider|
+        payload["providers"] << {
+          "environment" => "production",
+          "name"        => provider.name,
+          "type"        => provider.class.to_s,
+          "id"          => provider.id}.merge("alerts" => alerts_and_states(provider))
+      end
+      render_options(:alert_statuses, payload)
+    end
+
+    private
+
+    def alerts_and_states(provider)
+      result = []
+      provider.miq_alert_statuses.each do |alert_status|
+        alert = {
+          "id"            => alert_status.id,
+          "evaluated_on"  => alert_status.evaluated_on,
+          "link_text"     => alert_status.miq_alert.link_text,
+          "node_hostname" => alert_status.resource.name,
+          "description"   => alert_status.miq_alert.description,
+          "severity"      => alert_status.miq_alert.severity,
+          "states"        => add_alert_status_state(alert_status)
+        }
```

Comment:
can above be some sort of beautify/format helper method with app/models/miq_alert_status.rb ?